app/src/main/java/com/sopt/now/LoginActivity.kt

```diff
@@ -0,0 +1,55 @@
+package com.sopt.now
+
+import android.content.Intent
+import android.os.Build
+import android.os.Bundle
+import android.widget.Toast
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AppCompatActivity
+import com.sopt.now.databinding.ActivityLoginBinding
+
+class LoginActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityLoginBinding
+    private var data: SignUpData? = null
+
+    private val signupLauncher = registerForActivityResult(
+        ActivityResultContracts.StartActivityForResult()
+    ) {
+        val isSignup = it.data?.getBooleanExtra("isSignup", false) ?: false
+        if (it.resultCode == RESULT_OK && isSignup) {
+            data =
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU)
+                    it.data?.getSerializableExtra("data", SignUpData::class.java)
+                else it.data?.getSerializableExtra("data") as SignUpData
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        binding = ActivityLoginBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        with(binding) {
+            btnSignup.setOnClickListener { navigateToSignup() }
+            btnLogin.setOnClickListener { login() }
+        }
+    }
+
+    private fun navigateToSignup() =
+        Intent(this, SignupActivity::class.java).let { signupLauncher.launch(it) }
+
+    private fun login() {
+        if (validateLogin()) {
+            Toast.makeText(this, "로그인에 성공했습니다", Toast.LENGTH_SHORT).show()
+            Intent(this, MainActivity::class.java)
+                .putExtra("data", data)
```

Comment:
putExtra도 apply안에 넣을 수 있을거 같아

Reply:
putExtra를 apply안에 넣고 startActivity는 반환되는 객체값을 넣어주는 게 좋을 것 같아요!

```suggestion
       // 1번
       val intent = Intent(this, MainActivity::class.java).apply {
                putExtra("data", data)
               }
               startActivity(intent)
               
        // 2번 
               Intent(this, MainActivity::class.java).apply {
                putExtra("data", data)
               }.let {
                   startActivity(it)
               }
           // 3번
           Intent(this, MainActivity::class.java).apply {
                putExtra("data", data)
               }.let(::startActivity)
```